dbms/src/Core/SettingsCommon.h

```
@@ -423,20 +431,20 @@ class SettingsCollection
     public:
         reference(Derived & collection_, const MemberInfo & member_) : const_reference(collection_, member_) {}
         reference(const const_reference & src) : const_reference(src) {}
-        void setValue(const Field & value) { this->member->set_field(*const_cast<Derived *>(this->collection), value); }
-        void setValue(const String & value) { this->member->set_string(*const_cast<Derived *>(this->collection), value); }
-        void updateValue(const Field & value)
+        void setValue(const Field & value)
         {
-            if (!this->member->updateable)
+            if (this->member->is_immutable(*this->collection))
```

For consistency `field.setValue()` should throw an exception on change too.